### ISSUE TITLE
Refactor basic functions

### ISSUE THREAD

**Issue by benjamindeleener** [neuropoly/spinalcordtoolbox#538]
```sct_label_utils -t cubic-to-point``` is redondant with ```sct_label_utils -t label-vertebrae```. They could be merged.

```sct_crop_image``` should be integrated into ```sct_maths``` ?
Same for ```sct_crop_over_mask```.

Maybe a script called something like ```sct_tools``` would be better for grouping tools like ```sct_crop_image```, ```sct_copy_header```, etc.

**Comment by jcohenadad** [comment]
here are the suggested modifications for **sct_image**:
~~~
sct_image -i input1 -pad x,y,z
sct_image -i input1 -getorient
sct_image -i input -setorient RPI
sct_image -i input -setorient-data RPI
sct_image -i input -mcs -o output --> writes output_x, output_y, output_z
sct_image -i input_x,input1_y,input1_z -omc -o output
sct_image -i input -split dim{x,y,z,t} --> create input_T0000, input_T0001, ...
sct_image -i input1,input2 -concat dim{x,y,z,t} -o output
sct_image -i input_src -copy-header input_dest --> will copy source header into destination header
sct_image -i input_src -resample --> remove sct_resample
~~~

**Comment by jcohenadad** [comment]
create branch

**Comment by SaraDupont** [comment]
I created the branch **issue538_refactoring/**

**Comment by SaraDupont** [comment]
Options done on branch **issue538_refactoring/**:
~~~
sct_image -i input1 -pad x,y,z -o output
sct_image -i input -mcs -o output --> writes output_x, output_y, output_z
sct_image -i input_x,input1_y,input1_z -omc -o output
~~~

**Comment by SaraDupont** [comment]
Do we really want to include ```sct_resample``` to ```sct_image``` since it has a lot of parameteres/options (input as a refactoring actor, in mm or in voxels, interpolation, see doc below)  ?  

![doc_sct_resample](https://cloud.githubusercontent.com/assets/5857909/10313403/6b7615b6-6c1d-11e5-9037-b3018b565b84.png)

**Comment by jcohenadad** [comment]
sure, let's keep sct_resample separate for now. But please, could you fix the minor issues in the doc, e.g.,:
- missing "." between "expended" and "Default" for "-v"
- EXAMPLE should not show both -mm and -vox
- -f: Resampling factor in each dimension (x, y, z). Separate with "x"
- -mm: Resampling size in mm in each dimension (x, y, z). Separate with "x"
- -vox: Resampling size in number of voxels in each dimension (x, y, z). Separate with "x"
- -x {nn, linear, spline} Interpolation. (nn = nearest neighbor).
let's keep interpolation simple.
also please change "trilinear" to "linear".

**Comment by SaraDupont** [comment]
OK, it's done on the same branch.

**Comment by SaraDupont** [comment]
The output is now optional (to simplify the use of -getorient). A file name out is set by default in each function.

**Comment by SaraDupont** [comment]
All the tests are good but running the batch_processing, it affects somehow the segmentation of the dwi moco mean (which affects the template registration):
![dwi_seg](https://cloud.githubusercontent.com/assets/5857909/10376973/b0f27ee2-6dce-11e5-92a7-c51bfef0c5c9.gif)

**Comment by jcohenadad** [comment]
hum that's weird. What function is causing this difference?

**Comment by SaraDupont** [comment]
It's probably ```sct_dmri_moco``` : the dwi_moco_mean (that is segmented here) is not the same on dev or on this branch. I'm looking into the temporary files of dmri_moco (with the test_sct_dmri_moco function) : they are all the same on dev and on this branch except for dwi and dwi_mean ...

**Comment by jcohenadad** [comment]
hum, that's strange...
could you please create a test for split and concat to make sure that's indeed the problem?
It will also be useful for us to have a test for these functions in the future.
thanks!

**Comment by SaraDupont** [comment]
OK, I will. 
In fact the problem seems to come from ```sct_dmri_separate_b0_and_dwi```: 
- with dmri.nii.gz as input : same output image (b0, b0_mean, dwi, dwi_mean) on both branches
- with dmri_moco.nii.gz as input : dwi and dwi_mean aren't identical on both branches (with the same input image)

**Comment by jcohenadad** [comment]
ah! interesting. Could you take this opportunity to add an integrity test for ``sct_dmri_separate_b0_and_dwi``?
thanks!

**Comment by SaraDupont** [comment]
In  ```sct_dmri_separate_b0_and_dwi```, the dmri image is split along time, the time slices are separated in b0 and dwi, and the appropriate slices are concatenated to get the b0 and dwi image.
When refactoring the toolbox for the split_data and concat_data functions I tried to do (almost) all the calls in python directly instead of using sct.run(). So in this script I used a subsample of the list of images output by split_data() as input for concat_data() to get the b0 and dwi images. 

It appears that calling concat_data() directly in python does not give the same dwi result as calling it in command line with sct.run('sct_image -concat t ...'). I don't know why yet, (probably something to do with saving the images...) but it's a workaround that seems OK for me !
